### PR TITLE
feat(core): pdf viewer supports fit to page

### PR DESCRIPTION
**Pull request by fundon**
### What's Changed

* fits to page by default
* supports rendering pages of different sizes

https://github.com/user-attachments/assets/bf34e6d1-5546-4af1-a131-b801a67e9ace

### CODE REVIEW

**Comment by fundon** [comment]
* **#9307** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9307?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8812** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8812?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8812?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#8671** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8671?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8569** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8569?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 5 other dependent PRs ([#8615](https://github.com/toeverything/AFFiNE/pull/8615) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8615?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#8720](https://github.com/toeverything/AFFiNE/pull/8720) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8720?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#8764](https://github.com/toeverything/AFFiNE/pull/8764) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8764?utm_source=stack-comment-icon" target="_blank">Graphite</a> and 2 others)
* **#8542** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8542?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/toeverything?ref=merge-queue-instructions-in-stack-comment&prId=8812).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/frontend/core/src/components/attachment-viewer/pdf-viewer.tsx [hunk 1]
```diff
@@ -99,13 +103,23 @@ const PDFViewerInner = ({ pdf, state }: PDFViewerInnerProps) => {
     [pdf]
   );
 
+  const pagesContext = useMemo(() => {
+    const { width: w, height: h } = state.meta;
```
Comment (EYHN) [inline packages/frontend/core/src/components/attachment-viewer/pdf-viewer.tsx]: You need to handle PDF with different size pages

File: packages/frontend/core/src/utils/clipboard/index.ts [hunk 2]
```diff
@@ -30,9 +30,7 @@ export const copyLinkToBlockStdScopeClipboard = async (
 ) => {
   let success = false;
 
-  if (!clipboard) return success;
-
-  if (clipboardWriteIsSupported) {
+  if (clipboardWriteIsSupported && clipboard) {
```
Comment (forehalo) [inline packages/frontend/core/src/utils/clipboard/index.ts]: I know what do these changes try to fix, but separate them into different PR next time.

**Review by EYHN — APPROVED** [review]
